src/Illuminate/Collections/LazyCollection.php

```diff
@@ -1421,7 +1421,23 @@ public function sortKeysUsing(callable $callback)
     public function take($limit)
     {
         if ($limit < 0) {
-            return $this->passthru('take', func_get_args());
+            return new static(function () use ($limit) {
+                $limit = abs($limit);
+                $ringBuffer = [];
+                $position = 0;
+
+                foreach ($this as $key => $value) {
+                    $ringBuffer[$position] = [$key, $value];
+                    $position = ($position + 1) % $limit;
+                }
+
+                for ($i = 0; $i < $limit; $i++) {
+                    if (isset($ringBuffer[($position + $i) % $limit])) {
+                        [$key, $value] = $ringBuffer[($position + $i) % $limit];
+                        yield $key => $value;
+                    }
+                }
```

Comment:
@fuwasegu Might be worth getting rid of the extra work in each iteration:

```suggestion
                for ($i = 0, $end = min($limit, count($ringBuffer)); $i < $end; $i++) {
                    $pointer = ($position + $i) % $limit;
                    yield $ringBuffer[$pointer][0] => $ringBuffer[$pointer][1];
                }
```

Reply:
cool!
thank you  for your suggestion !

I have confirmed that the operation is equal in the environment at hand